ops/main.py

```diff
@@ -30,6 +30,13 @@ def _get_charm_dir():
 def _load_metadata(charm_dir):
     with open(charm_dir / 'metadata.yaml') as f:
         metadata = yaml.load(f, Loader=yaml.SafeLoader)
+
+    functions_meta = charm_dir / 'functions.yaml'
+    if not functions_meta.exists():
+        functions_meta = charm_dir / 'actions.yaml'
```

Comment:
Juju doesn't tell us if the newer functions API is supported and we can't tell from the version, since it's behind a feature flag.  However, from testing with the feature flag enabled, I've found:

* If `functions.yaml` is defined, both `actions.yaml` and `actions/` will be entirely ignored.
* If `actions.yaml` is defined `functions/` can be used but it must contain all of the action implementations, as `actions/` will be entirely ignored; otherwise, `actions/` will be used as per current behavior.

Of course, if the feature flag is not enabled or an older Juju is used, `functions.yaml` and `functions/` will be entirely ignored.

Reply:
After having more discussion with Ian, the intention is for the new API to always be available in 2.7+ regardless of feature flag (the flag just controls the CLI access to the feature).  Still, the actual use of the feature is driven by whether the dev uses `actions.yaml` or `functions.yaml`, and since the use of `actions/` vs `functions/` is handled by the framework, we should probably detect which filename they use, preferring `functions.yaml` like Juju does, and drive the symlinks based on that.

Reply:
For this specific block, what about this:

```python
    for filename in ('functions.yaml', 'actions.yaml'):
        functions_meta = charm_dir / filename
        if functions_meta.exists():
            with open(functions_meta) as f:
                metadata['functions'] = yaml.load(f, Loader=yaml.SafeLoader)
            break
```

Reply:
This is still based on whether the files exist. A file existing or not is defined solely by the charm.. if we deploy a charm in an old juju, the existence of functions.yaml doesn't mean anything. The ideal scenario is defining whether to support it or not based on juju, not on charm data that will exist independently.

Reply:
Just had a call with John, and we agreed that the easiest way out is to forbid both files, functions.yaml and actions.yaml, to exist at once in the charm. That's trivial to do and effective, and prevents bad charms anyway, as that's a bad idea. :+1:

Reply:
Per discussion, we want to consider having both styles to be a programmer error.